x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/ServiceUtils.java

```diff
@@ -91,6 +92,40 @@ public static <T> T removeAsType(Map<String, Object> sourceMap, String key, Clas
         }
     }
 
+    /**
+     * Remove the object from the map and cast to one of the expected types.
```

Comment:
Editing the comment to make it clear that the order of the types in `types` is important.

```suggestion
     * Remove the object from the map and cast to first assignable type in the expected types list.
```

Reply:
Makes a lot of sense, thanks!